cmd/kube-controller-manager/app/core.go

```diff
@@ -57,6 +57,14 @@ import (
 )
 
 func startServiceController(ctx ControllerContext) (bool, error) {
+
+	if ctx.Cloud == nil {
+		// don't return error when no cloud provider provided, just ourput a WARNING
+		glog.Warning("WARNING: no cloud provider provided, services of type LoadBalancer will fail." +
```

Comment:
This will log redundant info with https://github.com/kubernetes/kubernetes/blob/427dfd5ce1483ead29568bf7c05a98c4c0ad081c/pkg/controller/service/servicecontroller.go#L210, right?